src/widgets/helper/NotebookTab.cpp

```diff
@@ -381,6 +383,60 @@ void NotebookTab::setHighlightState(HighlightState newHighlightStyle)
     this->update();
 }
 
+void NotebookTab::setHighlightState(HighlightState newHighlightStyle,
+                                    ChannelView &channelViewSource,
+                                    const MessagePtr &message)
+{
+    if (this->isSelected())
+    {
+        return;
+    }
+
+    if (!this->highlightEnabled_ &&
+        newHighlightStyle == HighlightState::NewMessage)
+    {
+        return;
+    }
+
+    if (this->highlightState_ == newHighlightStyle ||
+        this->highlightState_ == HighlightState::Highlighted)
+    {
+        return;
+    }
+
+    auto *splitContainer =
+        dynamic_cast<SplitContainer *>(this->notebook_->getSelectedPage());
+    if (splitContainer != nullptr)
+    {
+        const auto &splits = splitContainer->getSplits();
+        for (const auto &split : splits)
+        {
+            auto &&filterIdsSource = channelViewSource.getFilterIds();
+            auto uniqueFilterIdsSource =
+                QSet(filterIdsSource.cbegin(), filterIdsSource.cend());
+            auto &&filterIdsSplit = split->getChannelView().getFilterIds();
+            auto uniqueFilterIdsSplit =
+                QSet(filterIdsSplit.cbegin(), filterIdsSplit.cend());
+
+            auto isSubset = []<typename T>(QSet<T> sub, QSet<T> super) {
+                return std::ranges::none_of(sub, [&super](const auto &subItem) {
+                    return !super.contains(subItem);
+                });
+            };
```

Comment:
1) I see, I used auto&& universal reference so that it always works, without any possible needless copies. In current impl it should be the same as you gave.
2) True, just wrote out my intent, to better spell it out.
3) sure